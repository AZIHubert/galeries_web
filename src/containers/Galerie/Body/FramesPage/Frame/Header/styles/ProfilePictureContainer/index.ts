import styled from 'styled-components';

const ProfilePictureImage = styled.div`
  border-radius: 50%;
  height: 26px;
  margin-right: 10px;
  overflow: hidden;
  width: 26px;
`;

export default ProfilePictureImage;
