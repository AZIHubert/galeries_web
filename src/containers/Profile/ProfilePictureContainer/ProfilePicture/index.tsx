import * as React from 'react';

import Modal from '#components/Modal';

import { UserContext } from '#contexts/UserContext';

import { ProfilePictureI } from '#helpers/interfaces';

import ModalProfilePicture from './ModalProfilePicture';

interface ProfilePictureComponentI {
  profilePicture: ProfilePictureI;
}

const ProfilePicture = ({
  profilePicture,
}: ProfilePictureComponentI) => {
  const [openModal, setOpenModal] = React.useState<boolean>(false);
  const { setUser } = React.useContext(UserContext);
  const handleClose = () => {
    setOpenModal(false);
  };
  const switchCurrent = () => {
    setUser((prevState) => {
      if (prevState) {
        const { currentProfilePicture } = prevState;
        if (currentProfilePicture && currentProfilePicture.id === profilePicture.id) {
          return {
            ...prevState,
            currentProfilePicture: null,
            currentProfilePictureId: null,
          };
        }
        return {
          ...prevState,
          currentProfilePicture: profilePicture,
          currentProfilePictureId: profilePicture.id,
        };
      }
      return null;
    });
  };
  return (
    <div
      data-testid='profilePicture'
    >
      <button
        data-testid='profilePictureButton'
        onClick={() => switchCurrent()}
      />
      <button
        onClick={() => setOpenModal((nextState) => !nextState)}
      >
        <img
          alt='profile picture'
          src={profilePicture.originalImage.signedUrl}
        />
      </button>
      <Modal
        open={openModal}
        handleClose={handleClose}
      >
        <ModalProfilePicture
          profilePicture={profilePicture}
        />
      </Modal>
    </div>
  );
};

export default ProfilePicture;
