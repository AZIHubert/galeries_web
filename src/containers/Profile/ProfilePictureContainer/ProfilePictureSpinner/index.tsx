import * as React from 'react';
import BounceLoader from 'react-spinners/BounceLoader';
import { CSSTransition } from 'react-transition-group';

import { ProfilePictureContext } from '#contexts/ProfilePictureContext';

import theme from '#helpers/theme';

import {
  Container,
  Fader,
  SpinnerContainer,
} from './styles';

const ProfilePictureSpinner = () => {
  const { isFetching } = React.useContext(ProfilePictureContext);

  return (
    <CSSTransition
      classNames='fade'
      in={isFetching}
      timeout={300}
      unmountOnExit
    >
      <Fader>
        <Container>
          <SpinnerContainer>
            <BounceLoader
              color={theme.colors.secondary}
              size={60}
            />
          </SpinnerContainer>
        </Container>
      </Fader>
    </CSSTransition>
  );
};

export default ProfilePictureSpinner;
