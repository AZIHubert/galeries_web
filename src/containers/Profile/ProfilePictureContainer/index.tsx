import * as React from 'react';

const ProfilPictureContainer = () => (
  <div>

  </div>
);

export default ProfilPictureContainer;
