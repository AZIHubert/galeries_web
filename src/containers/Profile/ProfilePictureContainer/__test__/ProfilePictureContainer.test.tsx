import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react';
import * as React from 'react';
import renderer from 'react-test-renderer';

import {
  UserContext,
  UserProvider,
} from '#contexts/UserContext';

import ProfilePictureContainer from '../index';

const newImage: ImageI = {
  bucketName: 'bucketName',
  fileName: 'fileName',
  format: 'jpg',
  height: 2340,
  id: '1',
  signedUrl: 'http://url.com',
  size: 100857,
  width: 1080,
};
const profilePictureOne: ProfilePictureI = {
  createdAt: new Date('2021-02-05 18:42:02.528+01'),
  cropedImage: newImage,
  id: '1',
  originalImage: newImage,
  pendingImage: newImage,
};
const profilePictureTwo: ProfilePictureI = {
  ...profilePictureOne,
  id: '2',
};
const newUser: UserI = {
  createdAt: new Date('2021-02-05 18:42:02.528+01'),
  currentProfilePicture: null,
  currentProfilePictureId: null,
  defaultProfilePicture: null,
  email: 'user@email.com',
  facebookId: null,
  googleId: null,
  id: '1',
  profilePictures: [profilePictureOne],
  role: 'admin',
  updatedAt: null,
  userName: 'user',
};

const Container = ({
  defaultUser = newUser,
}: {
  defaultUser?: UserI;
}) => {
  const { setUser, user } = React.useContext(UserContext);
  React.useEffect(() => {
    setUser(defaultUser);
  }, []);
  return (
    <>
      <ProfilePictureContainer />
      {user && user.currentProfilePictureId ? (
        <p
          data-testid='currentId'
        >
          {user.currentProfilePictureId}
        </p>
      ) : null}
    </>
  );
};

describe('ProfilePictureContainer', () => {
  afterEach(cleanup);
  it('renders without crashing', () => {
    const tree = renderer.create(<ProfilePictureContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render one profile picture', () => {
    const { getAllByTestId } = render(
      <UserProvider>
        <Container />
      </UserProvider>,
    );
    const profilePictures = getAllByTestId('profilePicture');
    expect(profilePictures.length).toBe(1);
  });
  it('should render Two profile picture', () => {
    const userWithTwoProfilePictures = {
      ...newUser,
      profilePictures: [
        profilePictureOne,
        profilePictureTwo,
      ],
    };
    const { getAllByTestId } = render(
      <UserProvider>
        <Container
          defaultUser={userWithTwoProfilePictures}
        />
      </UserProvider>,
    );
    const profilePictures = getAllByTestId('profilePicture');
    expect(profilePictures.length).toBe(2);
  });
  it('should set current profile id on click', () => {
    const { getByTestId } = render(
      <UserProvider>
        <Container />
      </UserProvider>,
    );
    const profilePictureButton = getByTestId('profilePictureButton');
    fireEvent.click(profilePictureButton);
    const currentId = screen.getByTestId('currentId');
    expect(currentId).toHaveTextContent(profilePictureOne.id);
  });
  it('should set to null profile id if is the same on click', () => {
    const { getByTestId } = render(
      <UserProvider>
        <Container />
      </UserProvider>,
    );
    const profilePictureButton = getByTestId('profilePictureButton');
    fireEvent.click(profilePictureButton);
    fireEvent.click(profilePictureButton);
    const currentId = screen.queryByTestId('currentId');
    expect(currentId).toBeNull();
  });
  it('should switch current id', () => {
    const userWithTwoProfilePictures = {
      ...newUser,
      profilePictures: [
        profilePictureOne,
        profilePictureTwo,
      ],
    };
    const { getAllByTestId } = render(
      <UserProvider>
        <Container
          defaultUser={userWithTwoProfilePictures}
        />
      </UserProvider>,
    );
    const secondProfilePictureButton = getAllByTestId('profilePictureButton')[1];
    fireEvent.click(secondProfilePictureButton);
    const currentId = screen.queryByTestId('currentId');
    expect(currentId).toHaveTextContent(profilePictureTwo.id);
  });
});
