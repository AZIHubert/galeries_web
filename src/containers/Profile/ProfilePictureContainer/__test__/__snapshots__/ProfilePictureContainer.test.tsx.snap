// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePictureContainer renders without crashing 1`] = `null`;
